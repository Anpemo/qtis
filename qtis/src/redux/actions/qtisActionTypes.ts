export default {
  USER_REGISTER: 'USER_REGISTER',
  PRODUCTS_LIST: 'PRODUCTS_LIST',
  SINGLE_PRODUCT: 'SINGLE_PRODUCT',
  CREATE_PRODUCT: 'CREATE_PRODUCT',
  REVIEWS_LIST: 'REVIEWS_LIST'
}
