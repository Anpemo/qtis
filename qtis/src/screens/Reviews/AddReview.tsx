import React, { useState } from 'react'
import {
  TouchableOpacity, View, Text, ImageBackground, KeyboardAvoidingView
} from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import { images } from '../../../constants'
import styles from './AddReviewStyles'
import { connect } from 'react-redux'
import { bindActionCreators } from 'redux'
import { createReview } from '../../redux/actions/qtisActionCreators'
import { AntDesign } from '@expo/vector-icons'
import { TextInput } from 'react-native-gesture-handler'

function AddReview ({ navigation, route, actions, user }: any) {
  const { productBarCode } = route.params
  const { userName, userPicture, _id } = user
  const { skinType } = user
  const [rating, setRating] = useState('')
  const [reviewText, setReviewText] = useState('')
  function shareReview () {
    actions.createReview({ rating: rating.replace(/,/g, '.'), reviewText, productBarCode, userId: _id, userName, userPicture, skinType })
    navigation.navigate('ProductDetail', productBarCode)
  }
  return (
    <SafeAreaView style={styles.container}>
      <ImageBackground source={images.backgroundReview} style={styles.backgroundImage}>
      <View style={styles.secondContainer}>
      <TouchableOpacity
      onPress={() => { navigation.goBack() }}
      style={styles.backIcon} >
        <AntDesign name="doubleleft" size={22}/>
      </TouchableOpacity>
        <KeyboardAvoidingView behavior={'padding'}>
      <View style={styles.inputsContainer}>
      <TextInput
          onChangeText={(number) => setRating(number.toString())}
          placeholder={'Rate from 1 to 5'}
          style={styles.rateInput}
          value={rating}
          keyboardType={'decimal-pad'}

        />
        <TextInput
          onChangeText={(text) => setReviewText(text)}
          placeholder={'Write your review'}
          style={styles.textInput}
          value={reviewText}
          multiline={true}
          numberOfLines={20}
        />
      </View>
        <TouchableOpacity
        onPress={() => shareReview()}
        style={styles.button}
        >
            <Text style={styles.buttonText}>SHARE</Text>
        </TouchableOpacity>
          </KeyboardAvoidingView>
      </View>
      </ImageBackground>
    </SafeAreaView>
  )
}
function mapStateToProps ({ productsReducer, userReducer }: any) {
  return {
    product: productsReducer.product,
    user: userReducer.user
  }
}

function mapDispatchToProps (dispatch: any) {
  return {
    actions: bindActionCreators({
      createReview
    }, dispatch)
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(AddReview)
