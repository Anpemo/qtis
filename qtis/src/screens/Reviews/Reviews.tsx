import React, { useEffect } from 'react'
import {
  View, Text, FlatList, Image
} from 'react-native'
import { SafeAreaView } from 'react-native-safe-area-context'
import { Rating } from 'react-native-ratings'
import { connect } from 'react-redux'
import { fetchReviews } from '../../redux/actions/qtisActionCreators'
import { bindActionCreators } from 'redux'
import styles from './ReviewsStyle'

function Reviews ({ reviews, actions, route, parameter }: any) {
  console.log('ruta', parameter)

  useEffect(() => {
    actions.fetchReviews(parameter)
  }, [])

  function renderReviews ({ item }: any) {
    (
        <View style={styles.reviewBox}>
          <View style={styles.userPictureBox}>
            <Image
            source={reviews.userPicture}
            style={styles.userPicture}
            key={1}
            />
          </View>
          <View style={styles.reviewContainer}>
            <Text style={styles.userName}>Angela Pedrero
            </Text>
            <Text style={styles.reviewText}>{item.reviewText}
            </Text>
          </View>
        </View>
    )
  }
  return (
        <SafeAreaView>
            <View style={styles.reviewsContainer}>
              <View style={styles.valorationContainer}>
                <Text style={styles.punctuation}>4,5</Text>
                <Rating
                type='star'
                ratingCount={5}
                readonly={true}
                fractions={4}
                imageSize={30}
                startingValue={4.5}
                ratingBackgroundColor={'black'}
                />
              </View>
              <FlatList
                  data = {reviews}
                  renderItem={renderReviews}
                  keyExtractor = {(item: any) => item.id}
                  horizontal={false}
                  style={styles.reviewsFlatList}

                  />
              </View>
        </SafeAreaView>
  )
}
function mapStateToProps ({ reviewsReducer }: any) {
  return {
    product: reviewsReducer.product
  }
}

function mapDispatchToProps (dispatch: any) {
  return {
    actions: bindActionCreators({
      fetchReviews
    }, dispatch)
  }
}
export default connect(mapStateToProps, mapDispatchToProps)(Reviews)
