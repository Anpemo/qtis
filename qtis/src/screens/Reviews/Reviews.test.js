import React from 'react'
import { cleanup, render, fireEvent } from '@testing-library/react-native'
import { Provider } from 'react-redux'
import configureStore from 'redux-mock-store'
import Reviews from './Reviews'
import * as actions from '../../redux/actions/qtisActionCreators'
jest.mock('../../redux/actions/qtisActionCreators')

const mockedNavigate = jest.fn()
jest.mock('@react-navigation/native', () => {
  return {
    ...jest.requireActual('@react-navigation/native'),
    useNavigation: () => ({
      navigate: mockedNavigate
    })
  }
})

describe('Given an Reviews function rendered from productDetail', () => {
  let componentFromProduct

  beforeEach(() => {
    const mockStore = configureStore()
    componentFromProduct = (
      <Provider store={mockStore({
        reviewsReducer: {
          reviews: [
            {
              brandName: 'Ives',
              userPicture: 'Crema facial',
              rating: 19,
              productCategory: 'creams',
              skinType: 'Normal',
              _id: 1
            },
            {
              brandName: 'Ives',
              userPicture: 'Crema facial',
              rating: 19,
              productCategory: 'creams',
              skinType: 'Normal',
              _id: 2

            }
          ]

        }
      })}>
        <Reviews parameter = {123123} />
        </Provider>
    )
    jest.spyOn(actions, 'fetchReviews').mockReturnValue({ type: '' })
  })

  afterEach(cleanup)

  describe('When rendering it from product && pressing to plus icon', () => {
    test('Then it will navigate', () => {
      const { getByTestId } = render(componentFromProduct)
      const navigateButton = getByTestId('navigateAddReview')
      fireEvent.press(navigateButton)
      expect(mockedNavigate).toHaveBeenCalled()
    })
  })
  describe('When rendering it from product && skinTypeButton', () => {
    test('Then it will setFiltered to false', () => {
      const renderedComponent = render(componentFromProduct)
      const skinTypeButton = renderedComponent.getAllByTestId('skinTypeButton')[0]
      fireEvent.press(skinTypeButton)
      expect(renderedComponent).toMatchSnapshot()
    })
    test('Then it will setFiltered to true', () => {
      const renderedComponent = render(componentFromProduct)
      const skinTypeButton = renderedComponent.getAllByTestId('skinTypeButton')[1]
      fireEvent.press(skinTypeButton)
      expect(renderedComponent).toMatchSnapshot()
    })
  })
  describe('When rendering it from product && reviews has no length', () => {
    test('Then it will render there are no reviews', () => {
      const mockStore = configureStore()

      componentFromProduct = (
        <Provider store={mockStore({
          reviewsReducer: {
            reviews: []
          }
        })}>
          <Reviews parameter = {123123} />
          </Provider>
      )

      const { getByTestId } = render(componentFromProduct)
      const container = getByTestId('valorationContainer')
      expect(container).toBeTruthy()
    })
  })
})
describe('Given an Reviews function rendered from profile', () => {
  let componentFromProfile
  beforeEach(() => {
    const mockStore = configureStore()

    componentFromProfile = (
    <Provider store={mockStore({
      reviewsReducer: {
        reviews: [
          {
            brandName: 'Ives',
            userPicture: 'Crema facial',
            rating: 19,
            productCategory: 'creams',
            _id: 1,
            skinType: 'Normal'
          },
          {
            brandName: 'Ives',
            userPicture: 'Crema facial',
            rating: 19,
            productCategory: 'creams',
            _id: 2,
            skinType: 'Normal'

          }
        ]

      }
    })}>
      <Reviews parameter = {'aaa'} />
      </Provider>
    )
    jest.spyOn(actions, 'fetchReviews').mockReturnValue({ type: '' })
  })
  afterEach(cleanup)

  describe('When rendering it', () => {
    test('Then profileProductName will be true', () => {
      const { getAllByTestId } = render(componentFromProfile)
      const productName = getAllByTestId('profileProductName')[0]
      expect(productName).toBeTruthy()
    })
  })
})
