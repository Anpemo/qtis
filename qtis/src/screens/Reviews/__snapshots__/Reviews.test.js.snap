// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given an Reviews function rendered from productDetail When rendering it from product && skinTypeButton Then it will setFiltered to false 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 3,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#141414",
          "fontFamily": "Montserrat",
          "fontSize": 20,
          "lineHeight": 20,
          "marginRight": 3,
        }
      }
    >
      19.0
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
        "marginVertical": 10,
        "paddingLeft": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#141414",
          "fontFamily": "Montserrat",
          "fontSize": 18,
        }
      }
    >
      Filter by skin type:
    </Text>
    <RCTScrollView
      data={
        Array [
          "None",
          "Normal",
          "Combination",
          "Oily",
          "Dry",
          "Sensitive",
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              None
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Normal
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Combination
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Oily
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Dry
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Sensitive
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 10,
          "paddingTop": 15,
          "width": "95%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratBold",
            "fontSize": 20,
          }
        }
      >
        REVIEWS
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="navigateAddReview"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 50,
          "marginTop": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
            "width": "95%",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "flex-start",
              "justifyContent": "center",
              "marginRight": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "Crema facial",
              }
            }
            style={
              Object {
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 100,
                "height": 70,
                "resizeMode": "cover",
                "width": 70,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "borderColor": "#D4A373",
              "borderRadius": 8,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "flex-start",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#D4A373",
                "fontFamily": "MontserratBold",
                "fontSize": 18,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#D4A373",
                "fontFamily": "MontserratBold",
                "fontSize": 18,
              }
            }
          >
            19
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "grey",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "evilicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "color": "#D4A373",
                "fontFamily": "Montserrat",
                "fontSize": 18,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
            "width": "95%",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "flex-start",
              "justifyContent": "center",
              "marginRight": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "Crema facial",
              }
            }
            style={
              Object {
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 100,
                "height": 70,
                "resizeMode": "cover",
                "width": 70,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "borderColor": "#D4A373",
              "borderRadius": 8,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "flex-start",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#D4A373",
                "fontFamily": "MontserratBold",
                "fontSize": 18,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#D4A373",
                "fontFamily": "MontserratBold",
                "fontSize": 18,
              }
            }
          >
            19
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "grey",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "evilicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "color": "#D4A373",
                "fontFamily": "Montserrat",
                "fontSize": 18,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;

exports[`Given an Reviews function rendered from productDetail When rendering it from product && skinTypeButton Then it will setFiltered to true 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "flex": 1,
      "width": "100%",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "flexDirection": "row",
        "justifyContent": "center",
        "marginBottom": 3,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#141414",
          "fontFamily": "Montserrat",
          "fontSize": 20,
          "lineHeight": 20,
          "marginRight": 3,
        }
      }
    >
      19.0
    </Text>
  </View>
  <View
    style={
      Object {
        "alignItems": "flex-start",
        "justifyContent": "center",
        "marginVertical": 10,
        "paddingLeft": 10,
        "width": "100%",
      }
    }
  >
    <Text
      style={
        Object {
          "color": "#141414",
          "fontFamily": "Montserrat",
          "fontSize": 18,
        }
      }
    >
      Filter by skin type:
    </Text>
    <RCTScrollView
      data={
        Array [
          "None",
          "Normal",
          "Combination",
          "Oily",
          "Dry",
          "Sensitive",
        ]
      }
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={true}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      style={
        Object {
          "flexDirection": "row",
          "width": "100%",
        }
      }
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              None
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Normal
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Combination
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Oily
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Dry
            </Text>
          </View>
        </View>
        <View
          onLayout={[Function]}
          style={
            Array [
              Object {
                "flexDirection": "row",
              },
              null,
            ]
          }
        >
          <View
            accessible={true}
            focusable={true}
            onClick={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              Object {
                "alignItems": "center",
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 50,
                "height": 30,
                "justifyContent": "center",
                "margin": 3,
                "opacity": 1,
                "width": 130,
              }
            }
            testID="skinTypeButton"
          >
            <Text
              style={
                Object {
                  "fontFamily": "Montserrat",
                  "fontSize": 16,
                }
              }
            >
              Sensitive
            </Text>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "alignItems": "center",
        "justifyContent": "center",
        "width": "100%",
      }
    }
  >
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "justifyContent": "space-between",
          "marginVertical": 10,
          "paddingTop": 15,
          "width": "95%",
        }
      }
    >
      <Text
        style={
          Object {
            "fontFamily": "MontserratBold",
            "fontSize": 20,
          }
        }
      >
        REVIEWS
      </Text>
      <View
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "opacity": 1,
          }
        }
        testID="navigateAddReview"
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "black",
                "fontSize": 30,
              },
              undefined,
              Object {
                "fontFamily": "anticon",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
        >
          
        </Text>
      </View>
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "marginBottom": 50,
          "marginTop": 10,
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
            "width": "95%",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "flex-start",
              "justifyContent": "center",
              "marginRight": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "Crema facial",
              }
            }
            style={
              Object {
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 100,
                "height": 70,
                "resizeMode": "cover",
                "width": 70,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "borderColor": "#D4A373",
              "borderRadius": 8,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "flex-start",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#D4A373",
                "fontFamily": "MontserratBold",
                "fontSize": 18,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#D4A373",
                "fontFamily": "MontserratBold",
                "fontSize": 18,
              }
            }
          >
            19
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "grey",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "evilicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "color": "#D4A373",
                "fontFamily": "Montserrat",
                "fontSize": 18,
              }
            }
          />
        </View>
      </View>
      <View
        style={
          Object {
            "flexDirection": "row",
            "marginBottom": 10,
            "width": "95%",
          }
        }
      >
        <View
          style={
            Object {
              "alignContent": "center",
              "alignSelf": "flex-start",
              "justifyContent": "center",
              "marginRight": 10,
            }
          }
        >
          <Image
            source={
              Object {
                "uri": "Crema facial",
              }
            }
            style={
              Object {
                "backgroundColor": "rgba(221, 161, 94, 0.3)",
                "borderRadius": 100,
                "height": 70,
                "resizeMode": "cover",
                "width": 70,
              }
            }
          />
        </View>
        <View
          style={
            Object {
              "alignItems": "flex-start",
              "borderColor": "#D4A373",
              "borderRadius": 8,
              "borderWidth": 1,
              "flex": 1,
              "justifyContent": "flex-start",
              "padding": 10,
            }
          }
        >
          <Text
            style={
              Object {
                "color": "#D4A373",
                "fontFamily": "MontserratBold",
                "fontSize": 18,
              }
            }
          />
          <Text
            style={
              Object {
                "color": "#D4A373",
                "fontFamily": "MontserratBold",
                "fontSize": 18,
              }
            }
          >
            19
            <Text
              allowFontScaling={false}
              style={
                Array [
                  Object {
                    "color": "grey",
                    "fontSize": 20,
                  },
                  undefined,
                  Object {
                    "fontFamily": "evilicons",
                    "fontStyle": "normal",
                    "fontWeight": "normal",
                  },
                  Object {},
                ]
              }
            >
              
            </Text>
          </Text>
          <Text
            style={
              Object {
                "backgroundColor": "#FFFFFF",
                "color": "#D4A373",
                "fontFamily": "Montserrat",
                "fontSize": 18,
              }
            }
          />
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
